tests/appsec/test_automated_user_and_session_tracking.py

```diff
@@ -0,0 +1,187 @@
+                        "operator": "exact_match",
+                    }
+                ],
+                "transformers": [],
+                "on_match": ["block"],
+            }
+        ],
+        "rules_data": [
+            {
+                "id": "blocked_users",
+                "type": "data_with_expiration",
+                "data": [{"value": "social-security-id", "expiration": 0}, {"value": "sdkUser", "expiration": 0}],
+            },
+        ],
+    },
+)
+
+
+@rfc("https://docs.google.com/document/d/1RT38U6dTTcB-8muiYV4-aVDCsT_XrliyakjtAPyjUpw")
+@features.user_monitoring
+@scenarios.appsec_runtime_activation
+class Test_Automated_User_Blocking:
+    def setup_user_blocking_auto(self):
+        rc.rc_state.reset().apply()
+
+        self.config_state_1 = rc.rc_state.set_config(*CONFIG_ENABLED).apply()
+        self.r_login = weblog.post("/login?auth=local", data=login_data(context))
+
+        self.config_state_2 = rc.rc_state.set_config(*BLOCK_USER).apply()
+        self.r_home_blocked = weblog.get("/", cookies=self.r_login.cookies,)
+
+    def test_user_blocking_auto(self):
+        assert self.config_state_1[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        assert self.r_login.status_code == 200
+
+        assert self.config_state_2[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        interfaces.library.assert_waf_attack(self.r_home_blocked, rule="block-users")
+        assert self.r_home_blocked.status_code == 403
+
+    def setup_user_blocking_sdk(self):
+        rc.rc_state.reset().apply()
+
+        self.config_state_1 = rc.rc_state.set_config(*CONFIG_ENABLED).apply()
+        self.config_state_2 = rc.rc_state.set_config(*BLOCK_USER).apply()
+        self.r_login_blocked = weblog.post(
+            "/login?auth=local&sdk_event=success&sdk_user=sdkUser", data=login_data(context)
+        )
+
+    def test_user_blocking_sdk(self):
+        assert self.config_state_1[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        assert self.config_state_2[rc.RC_STATE] == rc.ApplyState.ACKNOWLEDGED
+        interfaces.library.assert_waf_attack(self.r_login_blocked, rule="block-users")
+        assert self.r_login_blocked.status_code == 403
+
+
+@rfc("https://docs.google.com/document/d/1RT38U6dTTcB-8muiYV4-aVDCsT_XrliyakjtAPyjUpw")
+@features.user_monitoring
+class Test_Automated_Session_Tracking:
+    def setup_session_tracking(self):
+        self.r_create_session = weblog.get("/session/new")
+        self.r_home = weblog.get("/", cookies=self.r_create_session.cookies,)
+
+    def test_session_tracking(self):
+        assert self.r_create_session.status_code == 200
+        expected_session = self.r_create_session.text
+
+        assert self.r_home.status_code == 200
+        for _, trace, span in interfaces.library.get_spans(request=self.r_home):
+            meta = span.get("meta", {})
+            assert meta["usr.session_id"] == expected_session
```

Comment:
iiud "Session IDs must not be sent as part of the trace associated with the request, however, if a user chooses to do so through the User Tracking SDK, it should not be forbidden /prevented", so in case of automated session tracking we shouldnt see it in the trace no? just to the waf?

Reply:
Yes, you are right, I don't know where I got this from, I think the only way to test session ids is through blocking.